geomesa-spark/geomesa-spark-jts/src/main/scala/org/apache/spark/sql/jts/JTSTypes.scala

```diff
@@ -44,26 +52,401 @@ object JTSTypes {
   )
 }
 
-private [spark] class PointUDT extends AbstractGeometryUDT[Point]("point")
-object PointUDT extends PointUDT
+class PointUDT extends AbstractGeometryUDT[Point]("point"){
+
+  // parquet definition:
+  // group.id(GeometryBytes.TwkbPoint)
+  //  .required(PrimitiveTypeName.DOUBLE).named(GeometryColumnX)
+  //  .required(PrimitiveTypeName.DOUBLE).named(GeometryColumnY)
+
+  override val sqlType: StructType = StructType(Seq(
+    StructField("x", DataTypes.DoubleType, nullable = false),
+    StructField("y", DataTypes.DoubleType, nullable = false)
+  ))
+
+  override def serialize(obj: Point): InternalRow = {
+    if (obj == null) { null } else {
+      new GenericInternalRow(Array[Any](obj.getX, obj.getY))
+    }
+  }
+
+  override def deserialize(datum: Any): Point = {
+    if (datum == null) { null } else {
+      val row = ir(datum)
+      gf.createPoint(new Coordinate(row.getDouble(0), row.getDouble(1)))
+    }
+  }
+}
+
+case object PointUDT extends PointUDT
+
+class MultiPointUDT extends AbstractGeometryUDT[MultiPoint]("multipoint") {
+
+  // parquet definition:
+  // case ObjectType.MULTIPOINT =>
+  //   group.id(GeometryBytes.TwkbMultiPoint)
```

Comment:
i don't grok this stuff, but does this mean there some initial byte that needs to be included in the `sqlType` ?

Reply:
no, this is just how we define the field in our parquet implementation. it doesn't affect the serialization at all, and i don't think the name of the field really matters (although there are some rules around duplicate names), but we use the geometry type indicator byte as the name.